src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -230,6 +230,207 @@ .maxstack  4
             );
     }
 
+    [Fact]
+    [WorkItem(57731, "https://github.com/dotnet/roslyn/issues/57731")]
+    public void ListPattern_Codegen()
+    {
+        var source = @"
+public class X
+{
+    static int Test1(int[] x)
+    {
+        switch (x)
+        {
+            case [.., 1] and [1, ..]: return 0;
+        }
+
+        return 1;
+    }
+    static int Test2(int[] x)
+    {
+        switch (x)
+        {
+            case [2, ..] and [.., 1]: return 0;
+        }
+
+        return 3;
+    }
+    static int Test3(int[] x)
+    {
+        switch (x)
+        {
+            case [2, ..]: return 4;
+            case [.., 1]: return 5;
+        }
+
+        return 3;
+    }
+    static int Test4(int[] x)
+    {
+        switch (x)
+        {
+            case [2, ..]: return 4;
+            case [.., 1]: return 5;
+            case [6, .., 7]: return 8;
+        }
+
+        return 3;
+    }
+}
+";
+        var verifier = CompileAndVerify(new[] { source, TestSources.Index, TestSources.Range }, parseOptions: TestOptions.RegularWithListPatterns, options: TestOptions.ReleaseDll).VerifyDiagnostics();
+        AssertEx.Multiple(
+            () => verifier.VerifyIL("X.Test1", @"
```

Comment:
>VerifyIL [](http://example.com/codeflow?start=27&length=8)

Could you share result of decompiling IL to C#?

Reply:
<details><summary> Decompiled code from ListPattern_Codegen </summary>

```cs
private static int Test1(int[] x)
{
    if (x != null)
    {
        int num = x.Length;
        if (num >= 1 && x[num - 1] == 1 && x[0] == 1)
        {
            return 0;
        }
    }
    return 1;
}

private static int Test2(int[] x)
{
    if (x != null)
    {
        int num = x.Length;
        if (num >= 1 && x[0] == 2 && x[num - 1] == 1)
        {
            return 0;
        }
    }
    return 3;
}

private static int Test3(int[] x)
{
    if (x != null)
    {
        int num = x.Length;
        if (num >= 1)
        {
            if (x[0] == 2)
            {
                return 4;
            }
            if (x[num - 1] == 1)
            {
                return 5;
            }
        }
    }
    return 3;
}

private static int Test4(int[] x)
{
    if (x != null)
    {
        int num = x.Length;
        if (num >= 1)
        {
            int num2 = x[0];
            if (num2 == 2)
            {
                return 4;
            }
            int num3 = x[num - 1];
            if (num3 == 1)
            {
                return 5;
            }
            if (num >= 2 && num2 == 6 && num3 == 7)
            {
                return 8;
            }
        }
    }
}
```